doc/sample/getiteminfo.txt

```diff
@@ -13,28 +16,170 @@ prontera,156,179,6	script	test_getiteminfo	117,{
-			mes " getiteminfo(" + .@value + "," + .@id + ") = " + getiteminfo(.@value,.@id);
-		close;
+		break;
 	case 2:
 		mes "Please enter an english item name.";
 		input .@value$;
+		if(getiteminfo(.@value$,ITEMINFO_ID) == -1) { // Item doesn't exist, so end the script
+			mes "Unknown Item, please type the english name of an existing item.";
+			end;
+		}
 		mes "english item name: " + .@value$;
 		break;
 	case 3:
 		mes "Please enter an aegis item name.";
 		input .@value$;
+		if(getiteminfo(.@value$,ITEMINFO_ID) == -1) { // Item doesn't exist, so end the script
+			mes "Unknown Item, please type the aegis name of an existing item.";
+			end;
+		}
 		mes "aegis item name: " + .@value$;
 		break;
 	}
+	if(!.@value && .@value$ != "") // To prevent to duplicate the loop
+		.@value = getiteminfo(.@value$,ITEMINFO_ID);
 	mes "Current item info:";
-	for(.@id = 0; .@id <= 18; .@id++)
-		mes " getiteminfo(" + .@value$ + "," + .@id + ") = " + getiteminfo(.@value$,.@id);
+	for(.@id = 0; .@id < getarraysize(.IT_Info$); .@id++) {
+		mes .IT_Info$[.@id]+" ("+.@id+") = " + getiteminfo(.@value,.@id);
+		
+		if(.@id == ITEMINFO_TYPE) {
+			mes ">  "+.IT_Types$[getiteminfo(.@value,.@id)];
+		
+		} else if(.@id == ITEMINFO_GENDER) {
+			switch(getiteminfo(.@value,.@id)) {
+				case SEX_FEMALE: mes ">  Female only";
+				case SEX_MALE: mes ">  Male only";
+				case SEX_BOTH: mes ">  Both Gender";
+			}
+		} else if(.@id == ITEMINFO_LOCATIONS) {
+			.@loc = getiteminfo(.@value,.@id);
+			if(.@loc&EQP_COSTUME_HEAD_TOP) mes ">  EQP_COSTUME_HEAD_TOP";
+			if(.@loc&EQP_COSTUME_HEAD_MID) mes ">  EQP_COSTUME_HEAD_MID";
+			if(.@loc&EQP_COSTUME_HEAD_LOW) mes ">  EQP_COSTUME_HEAD_LOW";
+			if(.@loc&EQP_COSTUME_GARMENT) mes ">  EQP_COSTUME_GARMENT";
+			if(.@loc&EQP_SHADOW_ARMOR) mes ">  EQP_SHADOW_ARMOR";
+			if(.@loc&EQP_SHADOW_WEAPON) mes ">  EQP_SHADOW_WEAPON";
+			if(.@loc&EQP_SHADOW_SHIELD) mes ">  EQP_SHADOW_SHIELD";
+			if(.@loc&EQP_SHADOW_SHOES) mes ">  EQP_SHADOW_SHOES";
+			if(.@loc&EQP_SHADOW_ACC_R) mes ">  EQP_SHADOW_ACC_R";
+			if(.@loc&EQP_SHADOW_ACC_L) mes ">  EQP_SHADOW_ACC_L";
+			if(.@loc&EQP_ACC_L) mes ">  EQP_ACC_L";
+			if(.@loc&EQP_ACC_R) mes ">  EQP_ACC_R";
+			if(.@loc&EQP_SHOES) mes ">  EQP_SHOES";
+			if(.@loc&EQP_GARMENT) mes ">  EQP_GARMENT";
+			if(.@loc&EQP_HEAD_LOW) mes ">  EQP_HEAD_LOW";
+			if(.@loc&EQP_HEAD_MID) mes ">  EQP_HEAD_MID";
+			if(.@loc&EQP_HEAD_TOP) mes ">  EQP_HEAD_TOP";
+			if(.@loc&EQP_ARMOR) mes ">  EQP_ARMOR";
+			if(.@loc&EQP_HAND_L) mes ">  EQP_HAND_L";
+			if(.@loc&EQP_HAND_R) mes ">  EQP_HAND_R";
+			if(.@loc&EQP_ARMS) mes ">  EQP_ARMS";
+			if(.@loc&EQP_ACC_RL) mes ">  EQP_ACC_RL";
+			if(.@loc&EQP_SHADOW_ACC_RL) mes ">  EQP_SHADOW_ACC_RL";
+		
+		} else if(.@id == ITEMINFO_SUBTYPE) {
+			if(getiteminfo(.@value,ITEMINFO_TYPE) == IT_WEAPON || getiteminfo(.@value,ITEMINFO_TYPE) == IT_AMMO) {
+				mes ">  "+.IT_WSubTypes$[getiteminfo(.@value,.@id)];
+			}
+		}
+	}
 	close;
-}
+	
+	OnInit:
+	setarray .IT_Info$[0],
```

Comment:
There is no point saving data in a npc variable that will be used a few times

Reply:
While that's true, but one can just unload the NPC afterwards and the data is gone again, isn't it?
I also did it to "declutter" the script itself.
But if you prefer something like I did with the `ITEMINFO_LOCATIONS` then I will change it.

Reply:
So changed it now anyway:
![image](https://github.com/user-attachments/assets/ebfa734f-2ca8-4575-b7b5-a445fa3069fa)
![image](https://github.com/user-attachments/assets/58dd1cc1-2f24-4f95-b0f7-f15afca7ae31)

Once you satisfied with this, I will commit it.